### PR TITLE
feat(component): add private copy link button

### LINKED ISSUE #4485: ​ Add a copy link button at share menu
### Description

Add to bottom, [check design](https://www.figma.com/file/hVxJQtEqyv5xw1AbCzc50M/%F0%9F%97%92%EF%B8%8F-AFFiNE-2023?type=design&node-id=11287-19223&mode=design&t=wEPjqkLSNUIIkxKd-4)

Image

### PR DESCRIPTION
**Pull request by JimmFly**
Close #4485 

image

### CODE REVIEW

File: packages/component/src/components/share-menu/use-share-url.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
+import { useAFFiNEI18N } from '@affine/i18n/hooks';
+import { useCallback, useMemo } from 'react';
+
+import { toast } from '../../ui/toast';
+
+export enum UrlType {
+  Public = 'share',
+  UnPublic = 'workspace',
+}
```
Comment (pengx17) [inline packages/component/src/components/share-menu/use-share-url.ts:9]: The word UnPublic sounds strange. I guess you can just use `type UrlType = 'share' | 'workspace'`, without using enums.

**Review by pengx17 — APPROVED** [review]

File: packages/component/src/components/share-menu/use-share-url.ts (hunk 1 above)
Comment (JimmFly) [reply packages/component/src/components/share-menu/use-share-url.ts]: Made adjustments based on your suggestions

File: packages/component/src/components/share-menu/share-export.tsx [hunk 2]
```diff
@@ -25,6 +38,17 @@ export const ShareExport = () => {
         <div className={styles.descriptionStyle}>
           {t['com.affine.share-menu.ShareViaExportDescription']()}
         </div>
+        <Divider size="thinner" />
+        <div>
```
Comment (pengx17) [inline packages/component/src/components/share-menu/share-export.tsx:42]: Why wrap this button with a div?

Comment (JimmFly) [reply packages/component/src/components/share-menu/share-export.tsx:42]: Used to control its width

**Review by Svaney-ssman — APPROVED** [review]
👌🏻

**Pull request merged**